lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -402,6 +402,7 @@ def create_group_chain_methods(klass)
                 "count",
                 parameters: [
                   create_opt_param("column_name", type: "T.untyped", default: "nil"),
+                  create_block_param("block", type: "T.nilable(T.proc.params(record: T.untyped).returns(T.untyped))"),
```

Comment:
I don't know why groupchain `count` is different than regular `count` definition. I think it'll be good to take a closer look and be consistent if appropriate. Regular one uses overloaded sigs instead https://github.com/Shopify/tapioca/blob/91ad27251ad8c422f116cd0eafe9a5bd82ea7d70/lib/tapioca/dsl/compilers/active_record_relations.rb#L838.